From: lichenggang <lichenggang@uniontech.com>
Date: Mon, 01 Jun 2026 18:30:00 +0800
Subject: Fix sw64 CPU driver incompatible function pointer

The virCPUArchUpdate typedef (cpu.h:81-84) expects 3 parameters:
  (virCPUDef *guest, const virCPUDef *host, bool relative)

However virCPUsw64Update had an extra 4th parameter:
  (virCPUFeaturePolicy removedPolicy G_GNUC_UNUSED)

This causes a build failure with -Werror=incompatible-pointer-types:
  error: assignment to 'int (*)(virCPUDef *, const virCPUDef *,  bool)'
         from incompatible pointer type

Remove the extra parameter to match the expected signature.

Signed-off-by: lichenggang <lichenggang@uniontech.com>
---
 src/cpu/cpu_sw64.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/src/cpu/cpu_sw64.c b/src/cpu/cpu_sw64.c
index 1111111..2222222 100644
--- a/src/cpu/cpu_sw64.c
+++ b/src/cpu/cpu_sw64.c
@@ -56,8 +56,7 @@
 static int
 virCPUsw64Update(virCPUDef *guest,
                  const virCPUDef *host,
-                 bool relative ATTRIBUTE_UNUSED,
-                 virCPUFeaturePolicy removedPolicy G_GNUC_UNUSED)
+                 bool relative ATTRIBUTE_UNUSED)
 {
     g_autoptr(virCPUDef) updated = NULL;

--
2.43.0
